### ISSUE TITLE
Position window rule fails on bottom/rightmost workspaces

### ISSUE THREAD

**Issue by yonatan8070** [WayfireWM/wayfire#1264]
**Describe the bug**
I use sway-launcher-desktop with Alacritty as my launcher, and to launch it I'm using the following settings:
```ini
[command]
# Start your launcher
binding_launcher = <super> KEY_R
command_launcher = alacritty --class=launcher -e sway-launcher-desktop

[simple-tile]
tile_by_default = (!app_id is "launcher") & (floating is false)

[window-rules]
launcher_rule = on created if app_id contains "launcher" then set geometry 160 180 400 500
```
When it is launched on a workspace that is on the rightmost column, the launcher will be snapped the the left of the screen (X = 0) instead of the position the rule defines.

When it is launched on  a workspace that is on the bottom row, the launcher will be snapped the top of the screen (Y = 0)

When doing the above steps on the bottom right workspace, the window moves the the top left corner (X = 0 and Y = 0) 

**To Reproduce**
Use the above snippet and launch the launcher with Super+R on a workspace that is on the rightmost column or the bottom row (or both)

**Expected behavior**
Alacritty launches at the set position, instead of snapping to the top/left of the screen

**Screenshots or stacktrace**
Expected:
![Expected](https://user-images.githubusercontent.com/45665471/130276478-bfb6d686-9f5d-42a4-bdd3-9c5cf6b040cf.png)
Bottom workspace
![Bottom](https://user-images.githubusercontent.com/45665471/130276541-71bf6f66-ca9c-48f7-a161-f51f8983f0ce.png)
Right workspace
![Right](https://user-images.githubusercontent.com/45665471/130276878-3be24194-721d-4eb5-88d2-b7d71fc65fad.png)
Bottom right workspace
![Bottom Right](https://user-images.githubusercontent.com/45665471/130276972-6835ce64-0267-41b7-ba96-e2d58812c87f.png)
Expo
![expo](https://user-images.githubusercontent.com/45665471/130277109-3d946520-9e45-4a82-adc4-e3eca2152388.png)

**Wayfire version**
0.8.0-916a7eb4 (Jul 30 2021, branch 'HEAD')

**Comment by ammen99** [comment]
If you do not have this rule, what size would alacritty start with?

**Comment by yonatan8070** [comment]
With the rule commented out (and for regular Alacritty windows), Alacritty starts maximized, and with tiling disabled (because of the simple-tile rule)

**Comment by ammen99** [comment]
Alright, so yet another bug which would be trivially resolved by transactions (#1217). Since work there seems to need quite a lot of time, here's what happens:

1. When applying the rule, control flow reaches https://github.com/WayfireWM/wayfire/blob/master/plugins/window-rules/view-action-interface.cpp#L339
2. We first resize, but client needs time to respond, so view's geometry remains the same (e.g. maximized)
3. Then, we move, https://github.com/WayfireWM/wayfire/blob/master/plugins/window-rules/view-action-interface.cpp#L393 Because geometry is the same as before (resize hasn't happened yet), the plugin thinks that the view will go out of the workspace grid. This wouldn't be the case with the new size, but the code 'doesn't know it' yet.

Solution: make `_set_geometry` change both attributes (via `view->set_geometry()`) and do the clamping separately.

**Comment by ammen99** [comment]
This should be fixed in new Wayfire versions since transactions fix exactly the problem I described in my previous comment. Closing.
